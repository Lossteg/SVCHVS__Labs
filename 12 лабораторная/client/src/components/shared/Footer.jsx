import { Box, Container, Grid, Typography } from "@mui/material";
import { useTheme } from "@mui/material/styles";
import useMediaQuery from "@mui/material/useMediaQuery";
import { Link } from "react-router-dom";
import Logo from "./Logo";

function Footer() {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));

  return (
    <Box
      sx={{
        bgcolor: "primary.main",
        color: "white",
        p: 1,
        position: "fixed",
        bottom: 0,
        width: "100%",
      }}
    >
      <Container maxWidth="lg" style={{display: "flex", alignItems: "center"}}>
        <Logo className="logo-footer" />
        <Grid container spacing={isMobile ? 2 : 5}>
          <Grid item xs={12} sm={4}>
            <Link to="/about-us" className="router-link footer-link">
              <Typography>About us</Typography>
            </Link>
          </Grid>
          <Grid item xs={12} sm={4}>
            <Link to="/FAQ" className="router-link footer-link">
              <Typography>FAQ</Typography>
            </Link>
          </Grid>
          <Grid item xs={12} sm={4}>
            <Link to="/contacts" className="router-link footer-link">
              <Typography>Contact us</Typography>
            </Link>
          </Grid>
        </Grid>
      </Container>
    </Box>
  );
}

export default Footer;
