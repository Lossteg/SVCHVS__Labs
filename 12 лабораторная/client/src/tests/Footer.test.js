import React from "react";
import renderer from 'react-test-renderer';
import { render, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Footer from "../components/shared/Footer";

describe("Header component", () => {
  it("loads the user avatar image", async () => {
    const { container } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );

    // Ждем, пока изображение загрузится (может потребоваться время)
    const userAvatar = await waitFor(() => container.querySelector(".logo-footer"));

    expect(userAvatar).toBeTruthy();
  });
  it('should contain three links with correct text', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );

    const aboutUsLink = getByText('About us');
    const faqLink = getByText('FAQ');
    const contactUsLink = getByText('Contact us');

    expect(aboutUsLink).toBeInTheDocument();
    expect(faqLink).toBeInTheDocument();
    expect(contactUsLink).toBeInTheDocument();
  });
  it('Footer component matches snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Footer />
        </MemoryRouter>
      )
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });
});
