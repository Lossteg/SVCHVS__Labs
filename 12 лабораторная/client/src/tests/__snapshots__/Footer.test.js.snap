// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Footer component matches snapshot 1`] = `
<div
  className="MuiBox-root css-hp56ml"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <img
      alt="Copy-center-logo"
      className="logo-footer"
      src="CopyCenterLogo.png"
    />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1ld3b9g-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
      >
        <a
          className="router-link footer-link"
          href="/about-us"
          onClick={[Function]}
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            About us
          </p>
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
      >
        <a
          className="router-link footer-link"
          href="/FAQ"
          onClick={[Function]}
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            FAQ
          </p>
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
      >
        <a
          className="router-link footer-link"
          href="/contacts"
          onClick={[Function]}
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Contact us
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
